### ISSUE TITLE
Forced logouts are not making dreams come true

### ISSUE THREAD

**Issue by baka-kaba** [Awful/Awful.apk#336]
Mind if I create an issue for this? It's hard to diagnose and I thought we could put any clues we find in one place, and maybe work out what's causing it and how to trick it into working

If anyone ever sees the 'forums not available' page in a normal browser, it might be worth checking the source and see if there's a "notregistered" ID present. That forces the cookies to be cleared, but I get the feeling the error page has things like a LOG IN link too, even when you're logged in, so I wouldn't be surprised if that ID is there somewhere

**Comment by baka-kaba** [comment]
Well I'm definitely seeing a 'not logged in' page if I do two or three reloads in a row. I went to a pics thread, refreshed and waited until it started to load (progress bar visible), then refreshed again, and repeated that until it hit the logged out check in `AwfulError`.

It only took a couple of refreshes, and that number of loads seems to happen on occasion just from browsing normally. So it seems like either the outgoing requests are having problems, or the forums are blocking too many requests in quick succession?

**Comment by Sereri** [comment]
Maybe we should add a counter for that.

Logged Out -> Oh-really = true
 v
Logged In -> Oh really = false
OR
Logged Out -> Actually delete cookies

**Comment by baka-kaba** [comment]
Yeah actually I tried this - I made a little function that ignores these errors if you have a cookie, but after getting a few in a row it gives up and logs you out. I thought it would at least let the network cool off without getting in the way of new users or actual cookie issues, even if it doesn't fix the underlying problem

Anyway long story short, it doesn't seem to work. As soon as you hit that error, even without clearing the cookie, the next request will fail in the same way. Even if you wait a few seconds, next one gets the 'you are not registered' page. And the next. And eventually my function hits its sequential error limit and gives up (I reset it if a request doesn't hit that page)

So for whatever reason, once you get that page (for whatever reason), every subsequent request hits it (for whatever reason). I guess either the site is blocking you until you log back in, or something else is happening in the app. A lot of this network stuff is new to me, but as far as I can tell there are three lots of cookies:
* The cookies in the CookieStore that the DefaultHttpClient uses
* Some separate cookies WebView uses
* The stored values in the prefs
What I'm checking is the CookieStore set, which look fine - but is Volley using them? Maybe it's using another set internally, and they're getting wiped. I might be looking in the wrong place. Also I'm not sure if it matters, but the only two cookies we're using are the username and password ones - the session ones are never handled, not in the app's code anyway.

I'll push the branch I created to work on this, if you want to look at it - I refactored a bunch of stuff into NetworkUtils to try and keep it all in the same place, and added some log spam and checks. I'm guessing you don't want a pull request for that kind of thing, I'll just leave it on my repo
